spec/requests/custom_buttons_spec.rb

```diff
@@ -85,18 +87,20 @@
       request = {
         'action'    => 'edit',
         'resources' => [
-          { 'id' => cb.id.to_s, 'resource' => {'name' => 'updated 1', 'resource_action' => {'ae_namespace' => 'SYSTEM2'}}},
+          { 'id' => cb.id.to_s, 'resource' => {'name' => 'updated 1', 'resource_action' => {'ae_namespace' => 'SYSTEM2'}, 'visibility' => {'roles' => ['_ALL_']}}},
         ]
       }
       post(api_custom_buttons_url, :params => request)
 
       expected = {
         'results' => a_collection_including(
-          a_hash_including('id' => cb.id.to_s, 'name' => 'updated 1'),
+          a_hash_including('id' => cb.id.to_s, 'name' => 'updated 1', 'visibility' => {'roles' => ['_ALL_']}),
         )
       }
+      custom_button = CustomButton.find(response.parsed_body['results'].first["id"])
       expect(response).to have_http_status(:ok)
       expect(response.parsed_body).to include(expected)
+      expect(custom_button.visibility[:roles]).to eq(['_ALL_'])
```

Comment:
would prefer that we check the custom_button we have to start with, maybe just expect(cb.reload.visibility[:roles]).to ....

Reply:
I have addressed this in the last commit (5cb4355)

Reply:
commit revised further - 6d268dc